src/Plugin.Maui.Audio/AudioPlayer/AudioPlayer.macios.cs

```diff
@@ -27,29 +27,19 @@ public double Balance
 		set => player.Pan = (float)Math.Clamp(value, -1, 1);
 	}
 
-	public double Speed
+	public double Speed => player?.Rate ?? 0;
+
+	public void SetSpeed(double sp)
 	{
-		get => player.Rate;
-		set
+		// Rate property supports values in the range of 0.5 for half-speed playback to 2.0 for double-speed playback.
+		var speedValue = Math.Clamp((float)sp, 0.5f, 2.0f);
```

Comment:
These values don't match our properties/constants. We should try to use one source of values

Reply:
the public API exposes doubles while the AVPlayer expects floats. I was afraid this could lead to rounding mistakes.

Reply:
I was thinking it should be using the `minSpeed` and `maxSpeed` constants in the Clamp method call.

Reply:
On iOS I did not add a const float to the file as it was only used in one place. I had the same fear here, that using (float)minSpeed  (where minSpeed is of T double), could lead to a rounding mistake that would make minSpeed < 0.5, so that the method could fail.

Reply:
Sorry I'm reviewing on my phone and have been looking across 2 files 🤦‍♂️